src/defrag.c

```diff
@@ -1186,6 +1189,15 @@ void activeDefragCycle(void) {
 
     latencyEndMonitor(latency);
     latencyAddSampleIfNeeded("active-defrag-cycle",latency);
+
+update_metrics:
+    if (quit == 1) {
```

Comment:
i don't like the reliance on `quit`, this flag was intended to break out of two loops, not to signify that we're over the limit and have more work to do.
the variable that signifies that we have more work to do is `server.active_defrag_running`.
i we exit this function and that variable is enabled, it means we wanna be called again.